nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -108,6 +110,66 @@ public void shouldFailWhenMovingToColdStorageDocumentWithoutContent() {
         }
     }
 
+    @Test
+    public void shouldRetrieveDocumentBlobFromColdStorage() throws IOException {
+        DocumentModel documentModel = createDocument(true);
+
+        // move the blob to cold storage
+        ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+
+        // retrieve, which means initiate a request to restore the blob from cold storage
+        documentModel = ColdStorageHelper.retrieveContentFromColdStorage(session, documentModel.getRef(),
+                NUMBER_OF_DAYS_OF_AVAILABILITY);
+        transactionalFeature.nextTransaction();
+        documentModel = session.getDocument(documentModel.getRef());
```

Comment:
I would replace these 2 lines by `documentModel = session.save()` since we are not waiting for any asynchronous work, the retrieval is indeed a synchronous process.

Reply:
as explained on slack this is done to ensure that we will not have a false positive it's why i made `transactionalFeature.nextTransaction();` and i reload the document. we can replace 
`transactionalFeature.nextTransaction();` by `session.save()` but as usual we use `transactionalFeature.nextTransaction();` i used it. but if you think the use `session.save` is better i will modify it

Reply:
Sorry, yes I meant just
```
documentModel = session.getDocument(documentModel.getRef());
session.save()
```
But OK for ` transactionalFeature.nextTransaction()` :+1: